addons/web_editor/static/src/js/editor/odoo-editor/src/OdooEditor.js

```diff
@@ -4269,10 +4269,15 @@ export class OdooEditor extends EventTarget {
             if (fragment.hasChildNodes()) {
                 this._applyCommand('insert', fragment);
             }
-        } else if (files.length && targetSupportsHtmlContent) {
-            this.addImagesFiles(files).then(html => this._applyCommand('insert', this._prepareClipboardData(html)));
-        } else if (clipboardHtml && targetSupportsHtmlContent) {
-            this._applyCommand('insert', this._prepareClipboardData(clipboardHtml));
+        } else if ((files.length || clipboardHtml) && targetSupportsHtmlContent) {
+            const clipboardElem = this._prepareClipboardData(clipboardHtml);
+            if (files.length && !clipboardElem.querySelector('table')) {
+                // Images (not accompanied by HTML containing a table): added and wrapped.
```

Comment:
@dmo-odoo up :top:

Reply:
I feel like those two comments are too confused because the information is split in two parts and the only reason it is split in two parts is because you included the "what" part of the explanation that is visible in the code.

I suggest the following comment right above the line with `querySelector('table')`:
```js
// When copy pasting a table from the outside, a picture of the
// table can be included in the clipboard as an image file. In that
// particular case the html table is given a higher priority than
// the clipboard picture.
```

I would have liked to be more specific about the fact that the normal case is that images have precedence over the html but I don't recall exactly why we did that and I don't want to write a comment with false information.

Reply:
@dmo-odoo done.